AK/Math.h

```diff
@@ -741,15 +765,10 @@ constexpr T pow(T x, T y)
         return 0;
     if (y == 1)
         return x;
+
     int y_as_int = (int)y;
-    if (y == (T)y_as_int) {
-        T result = x;
-        for (int i = 0; i < fabs<T>(y) - 1; ++i)
-            result *= x;
-        if (y < 0)
-            result = 1.0l / result;
-        return result;
-    }
+    if (y == (T)y_as_int)
```

Comment:
While you are touching that line, can you make this use a `static_cast` instead?